### PR TITLE
Unit tests for init.go

### PR DESCRIPTION
**Pull request by VenkatRamaraju**
Added unit tests for v1alpha/init.go along with the suite tests for the v1 package.

### CODE REVIEW

**Review by laxmikantbpandhare — APPROVED** [review]

File: pkg/quarkus/v1alpha/init_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		successInitSubcommand.UpdateMetadata(testCliMetadata, &testSubcommandMetadata)
+
+	It("Check command name equality post function call", func() {
+			Expect(successInitSubcommand.commandName, testCliMetadata.CommandName)
+		})
+	})
+
+	Describe("BindFlags", func() {
+		flagTest := pflag.NewFlagSet("testFlag", -1)
+		// Need clarification on what to set ErrorHandlingNumber as
```
Comment (VenkatRamaraju) [inline pkg/quarkus/v1alpha/init_test.go:54]: Some suggestions on what to put as the ErrorHandlingNumber (currently -1) would be helpful.

File: pkg/quarkus/v1alpha/init_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		})
+
+		It("should set kind to an empty string", func() {
+			Expect(successInitSubcommand.kind, "")
+		})
+	})
+
+	Describe("InjectConfig", func() {
+		testConfig, _ := config.New(config.Version{Number: 3})
+		// Need clarification on what to set Version number as
```
Comment (VenkatRamaraju) [inline pkg/quarkus/v1alpha/init_test.go:85]: Here as well - suggestions on what to set as the "default" version number would be helpful.

**Comment by VenkatRamaraju** [comment]
Also, I did not write a test for the `Scaffold()` function ([here](https://github.com/operator-framework/java-operator-plugins/blob/main/pkg/quarkus/v1alpha/init.go#L119)). It only makes calls to external functions, so wasn't sure if I should trace all of them and write test cases for each one.

If you want me to dig into it and write a test for `Scaffold()` as well, I'll do that. Let me know.

**Comment by laxmikantbpandhare** [comment]
Everything is working perfect.

```
Running Suite: v1
=================
Random Seed: 1625769325
Will run 16 of 16 specs

••••••••••••••••
Ran 16 of 16 Specs in 0.000 seconds
SUCCESS! -- 16 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 1.138961774s
Test Suite Passed
```

File: pkg/quarkus/v1alpha/init_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+)
+
+var _ = Describe("v1", func() {
+	successInitSubcommand := &initSubcommand{domain: "testDomain"}
+	failureInitSubcommand := &initSubcommand{
+		domain:      "testDomain",
+		projectName: "?&fail&?",
+		commandName: "failureTest",
+	}
```
Comment (jmrodri) [inline pkg/quarkus/v1alpha/init_test.go:36]: Put in a `BeforeEach` block. that way each test has a fresh version that hasn't been affected by any changes from subsequent test.

File: pkg/quarkus/v1alpha/init_test.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+	Describe("UpdateMetadata", func() {
+		testCliMetadata := plugin.CLIMetadata{CommandName: "TestCommand"}
+		testSubcommandMetadata := plugin.SubcommandMetadata{}
+
+		It("Check command name inequality pre function call", func() {
+			Expect(failureInitSubcommand.commandName).NotTo(Equal(testCliMetadata.CommandName))
+		})
+
+		successInitSubcommand.UpdateMetadata(testCliMetadata, &testSubcommandMetadata)
```
Comment (jmrodri) [inline pkg/quarkus/v1alpha/init_test.go:46]: Should be part of the `It`

**Review by jmrodri — CHANGES_REQUESTED** [review]
Update the test similar to the other PR comments.

**Review by jmrodri — APPROVED** [review]
/lgtm

**Pull request merged**